Cannot import `grpc` module in Swift files when using CocoaPods

I am trying to use gRPC-Core 1.9.1 in a vanilla macOS Swift project with CocoaPods.

The pod module compiles just fine, but when I add `import grpc` to one of my Swift files, I get the following error:

    <module-includes>:8:9: note: in file included from <module-includes>:8:
    #import "Headers/support/log.h"
            ^
    /Users/daniel/Library/Developer/Xcode/DerivedData/grpc-import-test-btagazabopubmzbtzlbubflbuxqo/Build/Products/Debug/gRPC-Core/grpc.framework/Headers/support/log.h:22:10: error: include of non-modular header inside framework module 'grpc':     '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/clang/include/inttypes.h'
    #include <inttypes.h>
             ^
    /Users/daniel/Documents/Xcode/Mac/grpc-import-test/grpc-import-test/AppDelegate.swift:10:8: error: could not build Objective-C module 'grpc'
    import grpc
           ^

It appears that this error is specific to `inttypes.h`, as removing that one `include` line fixes the error. I suspect that the problem is that Clang for some reason resolves this to its internal header (which includes an `#include_next` directive), but does not recognize that as a header that does not to be part of the module. Would it be possible to remove the `#include <inttypes.h>` from that file?

FYI, this is my Podfile:

    target 'grpc-import-test' do
      use_frameworks!
    
      pod 'gRPC-Core', '~> 1.9'
      pod 'BoringSSL', '~> 9.1'
    end

The project is a brand-new, otherwise empty Swift project.

--- a/include/grpc/support/log.h
+++ b/include/grpc/support/log.h
@@ -21,13 +21,6 @@
 
 #include <grpc/impl/codegen/port_platform.h>
 
-/* On Apple platforms, including inttypes.h in a public header prevents gRPC
- * core to be built as framework. We rule out this inclusion on Apple platforms.
- */
-#if !defined(__APPLE__)
-#include <inttypes.h>
-#endif
-
 #include <stdarg.h>
 #include <stdlib.h> /* for abort() */
 

--- a/include/grpc/support/log.h
+++ b/include/grpc/support/log.h
@@ -21,7 +21,13 @@
 
 #include <grpc/impl/codegen/port_platform.h>
 
+/* On Apple platforms, including inttypes.h in a public header prevents gRPC
+ * core to be built as framework. We rule out this inclusion on Apple platforms.
+ */
+#if !defined(__APPLE__)
 #include <inttypes.h>
+#endif
+
 #include <stdarg.h>
 #include <stdlib.h> /* for abort() */
 

--- a/src/core/lib/gpr/log_linux.cc
+++ b/src/core/lib/gpr/log_linux.cc
@@ -32,6 +32,7 @@
 #include <grpc/support/log.h>
 #include <grpc/support/string_util.h>
 #include <grpc/support/time.h>
+#include <inttypes.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <string.h>

--- a/test/core/end2end/cq_verifier.cc
+++ b/test/core/end2end/cq_verifier.cc
@@ -18,6 +18,7 @@
 
 #include "test/core/end2end/cq_verifier.h"
 
+#include <inttypes.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <string.h>

--- a/test/core/end2end/tests/cancel_after_invoke.cc
+++ b/test/core/end2end/tests/cancel_after_invoke.cc
@@ -18,6 +18,7 @@
 
 #include "test/core/end2end/end2end_tests.h"
 
+#include <inttypes.h>
 #include <stdio.h>
 #include <string.h>
 

--- a/test/core/end2end/tests/cancel_before_invoke.cc
+++ b/test/core/end2end/tests/cancel_before_invoke.cc
@@ -18,6 +18,7 @@
 
 #include "test/core/end2end/end2end_tests.h"
 
+#include <inttypes.h>
 #include <stdio.h>
 #include <string.h>
 

--- a/test/core/end2end/tests/cancel_with_status.cc
+++ b/test/core/end2end/tests/cancel_with_status.cc
@@ -18,6 +18,7 @@
 
 #include "test/core/end2end/end2end_tests.h"
 
+#include <inttypes.h>
 #include <stdio.h>
 #include <string.h>
 

--- a/test/core/end2end/tests/negative_deadline.cc
+++ b/test/core/end2end/tests/negative_deadline.cc
@@ -18,6 +18,7 @@
 
 #include "test/core/end2end/end2end_tests.h"
 
+#include <inttypes.h>
 #include <stdio.h>
 #include <string.h>
 

--- a/test/core/gpr/mpscq_test.cc
+++ b/test/core/gpr/mpscq_test.cc
@@ -18,6 +18,7 @@
 
 #include "src/core/lib/gpr/mpscq.h"
 
+#include <inttypes.h>
 #include <stdlib.h>
 
 #include <grpc/support/alloc.h>

--- a/test/core/gpr/time_test.cc
+++ b/test/core/gpr/time_test.cc
@@ -21,6 +21,7 @@
 #include <grpc/support/log.h>
 #include <grpc/support/sync.h>
 #include <grpc/support/time.h>
+#include <inttypes.h>
 #include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>

--- a/test/core/util/test_config.cc
+++ b/test/core/util/test_config.cc
@@ -18,6 +18,7 @@
 
 #include "test/core/util/test_config.h"
 
+#include <inttypes.h>
 #include <signal.h>
 #include <stdbool.h>
 #include <stdio.h>

--- a/test/cpp/thread_manager/thread_manager_test.cc
+++ b/test/cpp/thread_manager/thread_manager_test.cc
@@ -16,6 +16,7 @@
  *is % allowed in string
  */
 
+#include <inttypes.h>
 #include <ctime>
 #include <memory>
 #include <string>

